core/src/main/java/com/squarespace/template/plugins/platform/CommerceUtils.java

```diff
@@ -140,6 +140,78 @@ public static JsonNode getLowestPriceAmongVariants(JsonNode item) {
     }
   }
 
+  public static JsonNode getPricingOptionsAmongLowestVariant(JsonNode item) {
+    ProductType type = getProductType(item);
+    JsonNode structuredContent = item.path("structuredContent");
+
+    switch (type) {
+      case PHYSICAL:
+      case SERVICE:
+        JsonNode variants = structuredContent.path("variants");
+        if (variants.size() == 0) {
+          return null;
+        }
+
+        JsonNode first = variants.get(0);
+        JsonNode moneyNode = isTruthy(first.path("onSale"))
+                ? first.path("salePriceMoney")
+                : first.path("priceMoney");
+
+        JsonNode pricingOptions = first.path("pricingOptions");
+
+        Decimal price = getAmountFromMoneyNode(moneyNode);
+
+        for (int i = 1; i < variants.size(); i++) {
+          JsonNode var = variants.get(i);
+          JsonNode currentMoneyNode = isTruthy(var.path("onSale"))
+                  ? var.path("salePriceMoney")
+                  : var.path("priceMoney");
+
+          Decimal current = getAmountFromMoneyNode(currentMoneyNode);
+          if (current.compare(price) < 0) {
+            price = current;
+            pricingOptions = var.path("pricingOptions");
+          }
+        }
+
+        return pricingOptions;
+
+      default:
+        return null;
+    }
+  }
+
+  public static JsonNode getSubscriptionMoneyFromPricingOptions(JsonNode pricingOptions, String subscriptionId) {
+    JsonNode chosenSubscription = null;
+
+    if (pricingOptions == null || pricingOptions.size() == 0) {
+      return DEFAULT_MONEY_NODE;
+    }
+
+    if (subscriptionId == null) {
+      return getMoneyAmountFromNode(pricingOptions.get(0));
+    }
+
+    for (int i = 0; i < pricingOptions.size(); i++) {
+      if (subscriptionId.equals(pricingOptions.get(i).path("productSubscriptionOptionId").asText())) {
```

Comment:
I'm confused here - this reads as though we're searching through pricingOptions to find ones that match a given subscription id? 🤔 Can you clarify overall what logic for this function is doing? I assumed it was getting the lowest discounted subscription price on the lowest priced variant, but after reading through this function I'm not sure.

Reply:
So this method is essentially just finding a `pricingOption` with a specific subscriptionId IF it is provided. `subscriptionId` is optional. So if it's not provided, we only return the first pricing option amongst the cheaper variant. We return the first pricing option because in NBF, we default the initial subscription to the first subscription in the list (Remember that merchant can go into composer and re-order this list).

Reply:
This is where we grab the list of pricingOptions for the cheapest variant: https://github.com/Squarespace/template-compiler/pull/52/files#diff-0f4b2b5b2c947bc1d756158308444fdab239501a935f531d3e0c8a28bbc64715R499. We either grab the first one from that list OR find the one by subscriptionId that is passed in as an optional argument